File: ios/DailyMemory/Domain/UseCase/Reminder/GetUpcomingRemindersUseCase.swift
import Foundation

/// Use case for getting upcoming reminders.
final class GetUpcomingRemindersUseCase {
    private let reminderRepository: ReminderRepository

    init(reminderRepository: ReminderRepository = ReminderRepositoryImpl()) {
        self.reminderRepository = reminderRepository
    }

    func execute(limit: Int = 5) async throws -> [Reminder] {
        return try await reminderRepository.getUpcoming(from: Date(), limit: limit)
    }
}

File: android/app/src/main/java/com/dailymemory/domain/usecase/reminder/GetUpcomingRemindersUseCase.kt
package com.dailymemory.domain.usecase.reminder

import com.dailymemory.domain.model.Reminder
import com.dailymemory.domain.repository.ReminderRepository
import kotlinx.coroutines.flow.Flow
import java.time.LocalDateTime
import javax.inject.Inject

/**
 * Use case for getting upcoming reminders.
 */
class GetUpcomingRemindersUseCase @Inject constructor(
    private val reminderRepository: ReminderRepository
) {
    operator fun invoke(limit: Int = 5): Flow<List<Reminder>> {
        return reminderRepository.observeUpcoming(LocalDateTime.now(), limit)
    }
}
